datastream-server-restli/src/test/java/com/linkedin/datastream/server/TestCoordinator.java

```diff
@@ -1155,6 +1155,46 @@ public void testDatastreamWithBYOT() throws Exception {
     zkClient.close();
   }
 
+  @Test
+  public void testInvokePostDataStreamStateChangeAction() throws Exception {
+    String testCluster = "testDatastreamCreateUpdateDelete";
+    String connectorType = "connectorType";
+
+    TestHookConnector connector1 = new TestHookConnector("connector1", connectorType);
+    Coordinator coordinator1 = createCoordinator(_zkConnectionString, testCluster);
+    coordinator1.addConnector(connectorType, connector1, new BroadcastStrategy(Optional.empty()),
+        false, new SourceBasedDeduper(), null);
+    coordinator1.start();
+    ZkClient zkClient = new ZkClient(_zkConnectionString);
+
+    // create datastream
+    Datastream[] list = DatastreamTestUtils.createAndStoreDatastreams(zkClient, testCluster, connectorType,
+        "datastream1");
+    Datastream datastream1 = list[0];
```

Comment:
With prev implementation this should have mattered but now we run this logic as part of the REST call, so in this method, this would be based on per datastream. The reason I added the test is that we have new public method. I have added it, you can see when we create datastreamResource we mention what coordinator to use.

Reply:
It was a very good testing scenario that verifies that  "Update on one datastream does not trigger event for another datastream".